clients/python/src/model_registry/core.py

```diff
@@ -17,30 +22,58 @@ class ModelRegistryAPIClient:
     def __init__(
         self,
         server_address: str,
-        port: int,
-        client_key: str | None = None,
-        server_cert: str | None = None,
-        custom_ca: str | None = None,
+        port: int = 443,
+        user_token: bytes | None = None,
+        custom_ca: bytes | None = None,
     ):
         """Constructor.
 
         Args:
             server_address: Server address.
-            port: Server port.
-            client_key: The PEM-encoded private key as a byte string.
-            server_cert: The PEM-encoded certificate as a byte string.
-            custom_ca: The PEM-encoded root certificates as a byte string.
+            port: Server port. Defaults to 443.
+            user_token: The PEM-encoded user token as a byte string. Defaults to envvar KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: The PEM-encoded root certificates as a byte string. Defaults to envvar CERT.
         """
-        config = MetadataStoreClientConfig()
-        config.host = server_address
-        config.port = port
-        if client_key is not None:
-            config.ssl_config.client_key = client_key
-        if server_cert is not None:
-            config.ssl_config.server_cert = server_cert
-        if custom_ca is not None:
-            config.ssl_config.custom_ca = custom_ca
-        self._store = MLMDStore(config)
+        if not user_token:
```

Comment:
code under here seems read confusing to me. are we saying when port is 443 it is secure? or having a ca_cert is secure? some code reorg could be better imo

Reply:
I provided two separate builders in the last commit, wdyt?